client/CPlayerInterface.cpp

```diff
@@ -1607,6 +1607,29 @@ void CPlayerInterface::objectRemoved(const CGObjectInstance * obj)
 	}
 }
 
+void CPlayerInterface::playerBlocked(int reason, bool start)
+{
+	if(reason == PlayerBlocked::EReason::UPCOMING_BATTLE)
+	{
+		if(CSH->howManyPlayerInterfaces() > 1 && LOCPLINT != this && LOCPLINT->makingTurn == false) 
+		{ 
+			//one of our players who isn't last in order got attacked not by our another player (happens for example in hotseat mode)
+			boost::unique_lock<boost::mutex> lock(eventsM); //TODO: copied from yourTurn, no idea if it's needed
+			LOCPLINT = this;
+			GH.curInt = this;
+			adventureInt->selection = nullptr;
+			adventureInt->setPlayer(playerID);
+			std::string msg = CGI->generaltexth->localizedTexts["adventureMap"]["playerAttacked"].String();
+			boost::replace_first(msg, "%s", cb->getStartInfo()->playerInfos.find(playerID)->second.name);
+			std::vector<std::shared_ptr<CComponent>> cmp;
+			cmp.push_back(std::make_shared<CComponent>(CComponent::flag, playerID.getNum(), 0));
+			makingTurn = true; //workaround for stiff showInfoDialog implementation
+			showInfoDialog(msg, cmp);
+			makingTurn = false;

```

Comment:
With those 3 lines you bypassed problem for this exact particular case, but all other notifications during battle still hidden from attacked player.